vnext/Microsoft.ReactNative/ReactApplication.h

```diff
@@ -4,15 +4,55 @@
 #pragma once
 
 #include "ReactApplication.g.h"
+#include <CppWinRTIncludes.h>
 #include "ReactNativeHost.h"
 
 namespace winrt::Microsoft::ReactNative::implementation {
 
-struct ReactApplication : ReactApplicationT<ReactApplication> {
-  using Super = ReactApplicationT<ReactApplication>;
+// NoDefaultCtorReactApplication_base is a copy of the generated ReactApplication_base
+// without the default constructor where it calls a factory for the base type.
+// This is done to fix the aggregation issue in types inheriting from the ReactApplication.
+// We call the factory in the ReactApplication constructor where we can pass correct
+// 'outer' interface.
+//
+// This class must match the generated ReactApplication_base.
+// It must be updated if the shape of generated ReactApplication_base is changed in future.
+// The only difference is that this class has no default constructor.
+template <typename D, typename... I>
+struct __declspec(empty_bases) NoDefaultCtorReactApplication_base
+    : implements<
+          D,
+          Microsoft::ReactNative::ReactApplication,
+          composable,
+          composing,
+          xaml::IApplicationOverrides,
+          xaml::IApplicationOverrides2,
+          I...>,
+      impl::require<D, xaml::IApplication, xaml::IApplication2, xaml::IApplication3>,
+      impl::base<D, xaml::Application>,
+      xaml::IApplicationOverridesT<D>,
+      xaml::IApplicationOverrides2T<D> {
+  using base_type = NoDefaultCtorReactApplication_base;
+  using class_type = Microsoft::ReactNative::ReactApplication;
+  using implements_type = typename NoDefaultCtorReactApplication_base::implements_type;
+  using implements_type::implements_type;
+  using composable_base = xaml::Application;
+
+  hstring GetRuntimeClassName() const {
+    return L"Microsoft.ReactNative.ReactApplication";
+  }
 
+ protected:
+  using dispatch = impl::dispatch_to_overridable<D, xaml::IApplicationOverrides, xaml::IApplicationOverrides2>;
+  auto overridable() noexcept {
+    return dispatch::overridable(static_cast<D &>(*this));
+  }
+};
+
+struct ReactApplication : NoDefaultCtorReactApplication_base<ReactApplication> {
```

Comment:
Why did the base type for this need to change? Was there something provided in the generated code that was conflicting with this override?

Reply:
The default constructor of the ReactApplicationT calls the factory for the XAML Application class. If I do not replace the base class, then the XAML Application is going to be created twice: one in the base default constructor and the second in the our constructor with the outer reference.

Ideally, the generated class must have two constructors or one with an optional outer parameter.

---
In reply to: [418151912](https://github.com/microsoft/react-native-windows/pull/4755#discussion_r418151912) [](ancestors = 418151912)

Reply:
Oh I see. Yes that makes sense.